mlflow/utils/proto_json_utils.py

```diff
@@ -291,25 +291,32 @@ def dataframe_from_parsed_json(decoded_input, pandas_orient, schema=None):
                 typemessage = "list"
             else:
                 typemessage = f"type {type(decoded_input)}"
-            raise MlflowBadScoringInputException(
+            raise MlflowInvalidInputException(
                 f"Dataframe split format must be a dictionary. Got {typemessage}."
             )
         keys = set(decoded_input.keys())
         missing_data = "data" not in keys
         extra_keys = keys.difference({"columns", "data", "index"})
         if missing_data or extra_keys:
-            raise MlflowBadScoringInputException(
+            raise MlflowInvalidInputException(
                 f"Dataframe split format must have 'data' field and optionally 'columns' "
                 f"and 'index' fields. Got {keys}.'"
             )
         try:
-            pdf = pd.DataFrame(
-                index=decoded_input.get("index"),
-                columns=decoded_input.get("columns"),
-                data=decoded_input["data"],
-            )
+            if schema is None:
+                pdf = pd.DataFrame(
+                    index=decoded_input.get("index"),
+                    columns=decoded_input.get("columns"),
+                    data=np.array(decoded_input["data"]),
+                )
+            else:
+                pdf = pd.DataFrame(
+                    index=decoded_input.get("index"),
+                    columns=decoded_input.get("columns"),
+                    data=decoded_input["data"],
+                )
```

Comment:
https://github.com/databricks/universe/pull/262773 Based on this PR, we update it.

Reply:
The newly added xgboost test doesn't pass on change`data=np.array(decoded_input["data"], dtype=object)`, as xgboost doesn't accept dtype=object

Reply:
We also can not set data=np.array(decoded_input["data"]) because it would convert data all to strings for example ```
>>> np.array([True, 1, "e"])
array(['True', '1', 'e'], dtype='<U21')
```